src/containers/Account/UpdateInfo/index.js

```diff
@@ -2,99 +2,123 @@ import {useCallback, useState} from 'react';
 import {useSelector} from 'react-redux';
 
 import Form from '@ra/components/Form';
-import Label from '@ra/components/Form/Label';
 import Input from '@ra/components/Form/Input';
 import {TextInput} from '@ra/components/Form/inputs';
+import {InputField} from '@ra/components/Form';
+import AccountPanel from '../AccountInfo';
 
 import store from 'store';
 import * as authActions from 'store/actions/auth';
 import Api from 'services/api';
 import Toast from 'services/toast';
 import useRequest from 'hooks/useRequest';
 
-import AccountPanel from '../AccountInfo';
 import styles from './styles.scss';
 
 const UpdateInfo = () => {
-    const {user} = useSelector(state => state.auth);
+    const {user} = useSelector((state) => state.auth);
     const {dispatch} = store;
-    const [userData, setUserData] = useState({
-        fullName: `${user.firstName} ${user.lastName}`,
-        email: user.email,
-        org: user.organization,
-        role: user.role,
-        success: true
-    });
 
-    const [error, setError] = useState(false);
-    const [infoState, updateInfo] = useRequest('/user/me/', {method: 'PATCH'});
-    
-    const handleSubmit = useCallback(async () => {
-        setError(null);
-        let firstName = '';
-        let lastName = '';
-        
-        if (userData.fullName.length >0 && userData.email.length>0 && userData.org.length>0 && userData.role.length>0) {
+    const [error, setError] = useState(null);
+    const [{loading}, updateInfo] = useRequest('/user/me/', {method: 'PATCH'});
+
+    const handleSubmit = useCallback(
+        async (formData) => {
+            setError(null);
+            const {fullName, email, organization, role} = formData;
+
+            const name = fullName.split(' ');
+            const firstName = name[0];
+            const lastName = fullName.substring(name[0].length).trim();
+
+            if (
+                user.firstName === firstName &&
+                user.lastName === lastName &&
+                user.email === email &&
+                user.organization === organization &&
+                user.role === role
+            ) {
+                Toast.show('NO Fields Changed !!', Toast.DANGER);
```

Comment:
I didn't mean that. Wanted this to be done in a bit elegant way. looks ugly it's better to remove this than the current implementation.

Reply:
Especially the comparison part. This will get a lot ugly if we wave lots of fields. FYI we have an object comparison function in our util.